### PR TITLE
HTTP2 - v18

### PR DESCRIPTION
**Pull request by jasonish**
Based on https://github.com/OISF/suricata/pull/5253/files.

- Rebased.
- Disables HTTP2 by default and marks it as experimental.
- Pull some fields out of the headers and place under the http object to be compatible with http logging.

Example output from S-V http2-basic:
https://gist.github.com/jasonish/21bd72ddd2c15b4434b7c45180d93f80#file-gistfile0-txt-L79

Example output from S-V http2-upgrade:
https://gist.github.com/jasonish/6636c079433caff7ede88b230e1149ee

suricata-verify-pr: 293

S-V is failing for some reason. Passes locally.

### CODE REVIEW

**Comment by victorjulien** [comment]
In the last record of your gist the headers are in the json record before the basic things like uri and method and such. I think it'd be nicer to reverse them.

**Comment by jasonish** [comment]
> In the last record of your gist the headers are in the json record before the basic things like uri and method and such. I think it'd be nicer to reverse them.

That would visually look better, but would require an extra pass over the headers, or using intermediate JSON builders. This is because we iterate the frames and log the headers as we do the iteration, which we have to do to pull out these fields.

**Comment by victorjulien** [comment]
The Debian builds fail due to compile errors around the derive use of hash. Any idea what is going on?

File: rust/src/http2/logger.rs [hunk 1]
```diff
@@ -0,0 +1,277 @@
...
+ * 02110-1301, USA.
+ */
+
+use super::http2::{HTTP2Frame, HTTP2FrameTypeData, HTTP2Transaction};
+use super::parser;
+use crate::jsonbuilder::{JsonBuilder, JsonError};
+use std;
+use std::collections::HashMap;
+
+#[derive(std::hash::Hash, PartialEq, Eq)]
```
Comment (victorjulien) [inline rust/src/http2/logger.rs:24]: changing this to just `Hash` (instead of `std::hash::Hash`) seems to fix it on Debian

**Comment by victorjulien** [comment]
Merged in #5284 with a few fixes. I also dropped the upgrade patches as they need more discussion.

**Pull request closed without merging**